assets/src/edit-story/components/movable/index.js

```diff
@@ -43,28 +42,126 @@ function Movable( {
+	} ) ) : [];
+
 	/**
 	 * Resets Movable once the action is done, sets the initial values.
 	 *
 	 * @param {Object} target Target element.
 	 */
 	const resetMoveable = ( target ) => {
-		frame.translate = [ 0, 0 ];
-		setStyle( target );
+		// @todo Improve this logic.
+		if ( targetList && targetList.length ) {
+			targetList.forEach( ( { ref }, i ) => {
+				frames[ i ].translate = [ 0, 0 ];
+				setStyle( ref, frames[ i ] );
+			} );
+		} else {
+			frame.translate = [ 0, 0 ];
+			setStyle( target, frame );
+		}
 		if ( moveable.current ) {
 			moveable.current.updateRect();
 		}
 	};
 
+	// @todo Perhaps split the group movable into a separate file.
+	// Moveable for group. Ensure that the targets list matches the selected elements list length.
+	if ( targetList && targetList.length ) {
+		return (
+			<Moveable
+				ref={ moveable }
+				target={ targetList.map( ( { ref } ) => ref ) }
+				draggable={ true }
+				resizable={ false }
+				rotatable={ true }
+				onDragGroup={ ( { events } ) => {
+					events.forEach( ( { target, beforeTranslate }, i ) => {
+						const sFrame = frames[ i ];
+						sFrame.translate = beforeTranslate;
+						setStyle( target, sFrame );
+					} );
+				} }
+				onDragGroupStart={ ( { events } ) => {
+					events.forEach( ( ev, i ) => {
+						const sFrame = frames[ i ];
+						ev.set( sFrame.translate );
+					} );
+				} }
+				onDragGroupEnd={ ( { targets } ) => {
+					const updatedElements = [];
+					// Set together updated elements.
+					targets.forEach( ( target, i ) => {
+						// @todo Improve this here.
+						updatedElements.push( { id: targetList[ i ].id, x: targetList[ i ].x + frames[ i ].translate[ 0 ], y: targetList[ i ].y + frames[ i ].translate[ 1 ] } );
+					} );
+					if ( updatedElements.length ) {
+						// Update the elements.
+						updateElementsByIds( updatedElements );
+					}
+					resetMoveable( null );
+				} }
+				onRotateGroupStart={ ( { events } ) => {
+					events.forEach( ( ev, i ) => {
+						const sFrame = frames[ i ];
+						ev.set( sFrame.rotate );
+					} );
+				} }
+				onRotateGroup={ ( { events } ) => {
+					events.forEach( ( { target, beforeRotate, drag }, i ) => {
+						const sFrame = frames[ i ];
+						sFrame.rotate = beforeRotate;
+						sFrame.translate = drag.beforeTranslate;
+						setStyle( target, sFrame );
+					} );
+				} }
+				onRotateGroupEnd={ ( { targets } ) => {
+					const updatedElements = [];
+					// Set together updated elements.
+					targets.forEach( ( target, i ) => {
+						updatedElements.push( {
+							id: targetList[ i ].id,
+							x: targetList[ i ].x + frames[ i ].translate[ 0 ],
+							y: targetList[ i ].y + frames[ i ].translate[ 1 ],
+							rotationAngle: frames[ i ].rotate,
+						} );
+					} );
+					if ( updatedElements.length ) {
+						// Update the elements.
+						updateElementsByIds( updatedElements );
+					}
+					resetMoveable( null );
+				} }
+				origin={ false }
+			/>
+		);
+	}
+
+	const frame = {
```

Comment:
Could you please place it higher since some of the functions above use it?

Reply:
I'm actually thinking of moving the Moveable for handling single vs group moving into separate files for testability and maintaining reasons. Kept this purposefully near the single Moveable. (Sorry, still a draft!)